File: src/main/java/com/sparetimedevs/questlog/user/UserResourceProcessor.java
package com.sparetimedevs.questlog.user;

import com.sparetimedevs.questlog.userpassword.UserPasswordController;
import org.springframework.hateoas.Resource;
import org.springframework.hateoas.ResourceProcessor;
import org.springframework.stereotype.Component;

import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;

@Component
public class UserResourceProcessor implements ResourceProcessor<Resource<User>> {

	@Override
	public Resource<User> process(Resource<User> resource) {
		resource.add(linkTo(UserPasswordController.class).withRel("save_password"));
		return resource;
	}
}

File: src/main/java/com/sparetimedevs/questlog/user/UserResourceProcessor.kt
package com.sparetimedevs.questlog.user

import com.sparetimedevs.questlog.login.LoginController
import com.sparetimedevs.questlog.userpassword.UserPasswordController
import org.springframework.hateoas.Resource
import org.springframework.hateoas.ResourceProcessor
import org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo
import org.springframework.stereotype.Component

@Component
class UserResourceProcessor : ResourceProcessor<Resource<User>> {

	override fun process(resource: Resource<User>): Resource<User> {
		resource.add(linkTo(UserPasswordController::class.java).withRel("save_password"))
		resource.add(linkTo(LoginController::class.java).withRel("login"))
		return resource
	}
}
